File: Examples/Node.js/Slides/GetSlideCountUsingThirdPartyStorage.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var SlidesApi = require('asposeslidescloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Slides API SDK
var slidesApi = new SlidesApi(config);

// Set input file name
var name =  "sample-input.pptx";
var slideIndex = 1;

var versionId = null;
var storage = "AsposeDropboxStorage";
var folder = null;

try {
// Upload file to 3rd party cloud storage
storageApi.PutCreate(name, versionId, storage, data_path + name , function(responseMessage) {

	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Slides Cloud SDK API to get PowerPoint document slide list
	slidesApi.GetSlidesSlidesList(name, folder, storage, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			var count = responseMessage.body.Slides.SlideList.length;
			console.log("Total Slides :: " + count);
			});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1

File: Examples/Perl/Slides/GetSlideCountUsingThirdPartyStorage.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposeSlidesCloud::SlidesApi;
use AsposeSlidesCloud::ApiClient;
use AsposeSlidesCloud::Configuration;

my $configFile = '../Config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposeSlidesCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeSlidesCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposeSlidesCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Slides API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $slidesApi = AsposeSlidesCloud::SlidesApi->new();

# Set input file name
my $name = 'sample-input.pptx';
my $slideIndex = 1;
my $storage = 'AsposeDropboxStorage';

# Upload file to 3rd party cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name, storage=>$storage);

# Invoke Aspose.Slides Cloud SDK API to get PowerPoint document slide list                                   
$response = $slidesApi->GetSlidesSlidesList(name => $name, slideIndex=>$slideIndex, storage=>$storage);

if($response->{'Status'} eq 'OK'){
	print "\n Total Slides :: ". scalar(@{$response->{'Slides'}->{'SlideList'}});
}
#ExEnd:1